csc2/macc_so.c

```diff
@@ -1238,6 +1238,18 @@ void key_piece_add(char *buf,
     macc_globals->workkeypieceflag = 0;
 }
 
+/* timezone not supported for now */
+static inline int valid_datetime(const char *str)
+{
+    int year, month, day, hour, min, sec;
+    int rc = sscanf(str, "%04d-%02d-%02dT%02d:%02d:%02d",                                                                   
+            &year, &month, &day, &hour, &min, &sec);
+    if (rc != 6) {
+        return 0;
+    }
+    return 1;
```

Comment:
When validating date/time values, do we normally have range checks for valid months, days, etc?  e.g. To prevent things like Feb 31st or a 13th month, etc.

Reply:
We need a more complete validator, that's true. I was hoping we already have it, somewhere in the code.

Reply:
you don't really need year month etc here. Consider using optional assignment?

Reply:
We allow denormalized values.